### PR TITLE
Add RepositoryClient.listPullRequestsForCommit

### PR DESCRIPTION
**Pull request by dziemba**
See https://docs.github.com/en/enterprise-server@3.2/rest/commits/commits#list-pull-requests-associated-with-a-commit

### CODE REVIEW

**Review by sathish-kumar-subramani — COMMENTED** [review]
Looks good.
Since this API responds with first 30 prs, should we handle this with pagination?
Like https://github.com/spotify/github-java-client/blob/e19601c2d170a5036be67e7ac19ea8258a952d3b/src/main/java/com/spotify/github/v3/clients/RepositoryClient.java#L258-L263


_note: But we do not handle pagination for most of the other APIs as well._

**Comment by dziemba** [comment]
Yes I think we don't need this and don't do pagination anywhere else. Let's keep it simple for now and add it if we need it later. I don't think it's ever needed in this case: will there ever be more than 30 PRs containing the same commit? Probably not 😉

File: src/main/java/com/spotify/github/v3/clients/RepositoryClient.java [hunk 1]
```diff
@@ -49,9 +51,11 @@
 import java.lang.invoke.MethodHandles;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionException;
+import javax.ws.rs.core.HttpHeaders;
```
Comment (vchernodon) [inline src/main/java/com/spotify/github/v3/clients/RepositoryClient.java:58]: There is an alternative class which we could use `com.google.common.net.HttpHeaders`

File: src/main/java/com/spotify/github/v3/clients/RepositoryClient.java [hunk 2]
```diff
@@ -272,6 +277,22 @@ public CompletableFuture<List<CommitItem>> listCommits() {
...
+   * List pull requests that contain the given commit.
+   *
+   * @param sha commit sha
+   * @return pull requests
+   */
+  public CompletableFuture<List<PullRequestItem>> listPullRequestsForCommit(final String sha) {
+    final String path = String.format(COMMIT_PULL_REQUESTS_SHA_URI_TEMPLATE, owner, repo, sha);
+
+    // As of GHE 3.2, this feature is still in preview, so we need to add the extra header.
+    // https://developer.github.com/changes/2019-04-11-pulls-branches-for-commit/
```
Comment (vchernodon) [inline src/main/java/com/spotify/github/v3/clients/RepositoryClient.java:290]: Great addition.

File: src/test/java/com/spotify/github/v3/clients/RepositoryClientTest.java [hunk 3]
```diff
@@ -171,6 +174,18 @@ public void listCommits() throws Exception {
...
   }
 
+  @Test
+  public void listPullRequestsForCommit() throws Exception {
+    final CompletableFuture<List<PullRequestItem>> fixture =
+        completedFuture(
+            json.fromJson("[" + getFixture("../prs/pull_request_item.json") + "]", LIST_PR_TYPE_REFERENCE));
+    when(github.request(eq("/repos/someowner/somerepo/commits/thesha/pulls"), eq(LIST_PR_TYPE_REFERENCE), any()))
+        .thenReturn(fixture);
+    final List<PullRequestItem> commits = repoClient.listPullRequestsForCommit("thesha").get();
```
Comment (vchernodon) [inline src/test/java/com/spotify/github/v3/clients/RepositoryClientTest.java:184]: Does it return commits? Or pull requests? if the latter is true, should be renamed to `pullRequests`? (as list)

Comment (dziemba) [reply src/test/java/com/spotify/github/v3/clients/RepositoryClientTest.java]: fixed

File: src/main/java/com/spotify/github/v3/clients/RepositoryClient.java (hunk 1 above)
Comment (dziemba) [reply src/main/java/com/spotify/github/v3/clients/RepositoryClient.java:58]: I just copied this from other classes in this project, doesn't matter much I think 🤷

**Review by vchernodon — APPROVED** [review]

**Pull request merged**